torchtune/datasets/_finetune.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Also since this is BC breaking, what is our plan to deprecate the old APIs? 2 release cycles like Core? Can this RFC include a warning in all the old datasets?

Reply:
Need to think about this more... do you have any suggestions for the rollout? The dataset builders will remain the same so users shouldn't need to change their configs. Especially if we keep tokenizer as a required argument and make model transform optional, it's only the underlying abstractions changing.